Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -476,6 +477,42 @@ fileprivate extension Array<AnnotatedTestItem> {
     }
     return result
   }
+
+  func prefixTestsWithModuleName(workspace: Workspace) async -> Self {
+    return await self.asyncMap({
+      // If the module name can't be determined we return the test item without a prefixed id.
+      guard let moduleName = await self.moduleName(from: workspace, for: $0.testItem.location.uri) else {
+        return $0
+      }
+      var newTest = $0.testItem
+      newTest.id = "\(moduleName).\(newTest.id)"
+      newTest.children = await prefixTestsWithModuleName(workspace: workspace, newTest.children)
+      return AnnotatedTestItem(testItem: newTest, isExtension: $0.isExtension)
+    })
+  }
+
+  private func prefixTestsWithModuleName(workspace: Workspace, _ tests: [TestItem]) async -> [TestItem] {
+    return await tests.asyncMap({
+      guard let moduleName = await self.moduleName(from: workspace, for: $0.location.uri) else {
+        return $0
+      }
+
+      var newTest = $0
+      newTest.id = "\(moduleName).\(newTest.id)"
+      newTest.children = await prefixTestsWithModuleName(workspace: workspace, newTest.children)
+      return newTest
+    })
+  }
+
+  private func moduleName(from workspace: Workspace, for uri: DocumentURI) async -> String? {
+    guard let configuredTarget = await workspace.buildSystemManager.canonicalConfiguredTarget(for: uri) else {
+      return nil
+    }
+    // If for whatever reason we can't get a module name from the build system, fall back
+    // to using the targetID as this would be used when there is no command line arguments
+    // to define the module name as something other than the targetID.
+    return await workspace.buildSystemManager.moduleName(for: uri, in: configuredTarget) ?? configuredTarget.targetID
```

Comment:
Is it a valid assumption that if there is no module name available we can use the `targetID`?

Reply:
I would prefer to return `nil` here if there is no module name since that will allow us to still run tests. If we picked the wrong module name, no test would be executed for that test ID. 

Was there a specific reason why you added the fallback?

Reply:
This was covering off the obj-c case, which it was doing implicitly (and incorrectly, since the module name could still be different than the target name). I've updated the argument parsing to include obj-c.